github/resource_github_team_sync_group_mapping_test.go

```diff
@@ -0,0 +1,97 @@
+package github
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/google/go-github/v29/github"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+)
+
+func TestAccGithubTeamSyncGroupMapping_basic(t *testing.T) {
+	var team github.Team
+
+	rn := "github_team.foo"
+	randStringTeam := acctest.RandStringFromCharSet(10, acctest.CharSetAlphaNum)
+	teamName := fmt.Sprintf("tf-acc-test-%s", randStringTeam)
+	randStringGroup := acctest.RandStringFromCharSet(10, acctest.CharSetAlphaNum)
+	groupID := acctest.RandString(5)
+	groupName := fmt.Sprintf("tf-acc-test-group-%s", randStringGroup)
+	description := "Terraform acc test group"
+	updatedDescription := "Terraform acc test group - updated"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckGithubTeamSyncGroupMappingDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccGithubTeamSyncGroupMappingConfig(teamName, groupID, groupName, description),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGithubTeamExists(rn, &team),
+					resource.TestCheckResourceAttrSet(rn, "groups"),
+					resource.TestCheckResourceAttr(rn, "groups.0.group_id", groupID),
+					resource.TestCheckResourceAttr(rn, "groups.0.group_name", groupName),
+					resource.TestCheckResourceAttr(rn, "groups.0.description", description),
+				),
+			},
+			{
+				Config: testAccGithubTeamSyncGroupMappingConfig(teamName, groupID, groupName, updatedDescription),
```

Comment:
I think we might want to do a few steps here updating the actual groups.  If we can make an array of maybe 2 GitHub groups, then modify it to 1, and/or maybe even 0, just to make sure it's doing that update correctly, I think that might be a good check as well as updating the content of the groups.